### ISSUE TITLE
How to remove an endpoint or operation

### ISSUE THREAD

**Issue by deiteris** [aml-org/amf#1460]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [ ] bug report
  - [ ] feature request
  - [x] question
  - [ ] other

I was reading the documentation and noticed that it only mentions creating new endpoints, operations and populate them with data, but never seen removing existing ones. Is there a method for that?

I'm trying to remove endpoints and/or their operations by specific condition, but can't figure out how to do this. Objects seem to be immutable when I'm trying to re-assign them and I don't see any suitable method.

So far, I've managed to partially achieve my goal by collecting and replacing endpoints and operations with the example below. But maybe there's more convenient way to do this?
```js
const client = amf.RAMLConfiguration.RAML10()
                .baseUnitClient();

const parsingResult = await client.parseDocument(`file://${sourceFile}`);

const transformationResult = client.transform(parsingResult.baseUnit, amf.PipelineId.Default);

const endpoints = []
for (const endpoint of transformationResult.baseUnit.encodes.endPoints) {
    const operations = []
    for (const operation of endpoint.operations) {
        if (_includesRole(operation)) { // Looks up a specific Custom Domain Property and includes if found
            operations.push(operation)
        }
    }
    if (!operations.length) {
        continue
    }
    endpoint.withOperations(operations)
    endpoints.push(endpoint)
}
transformationResult.baseUnit.encodes.withEndPoints(endpoints)
```

* **Please tell us about your environment:**
  
  - AMF Version: 5.0.9
  - AMF Distribution: JS 
  - Operating System: Windows 10 21H2

**Comment by tomsfernandez** [comment]
Hi @deiteris!. We don't provide a way to filter out endpoints or operations natively in our interface. You must do what you showed us:
- extract desired endpoints and operations from the baseUnit
- re-add them with `withOperations` or `withEndpoints`

We do this internally also. Our interface would grow a lot if we did this for every entity and if we only did it for some it wouldn't be coherent.

I'm closing this issue but if you have questions we can keep discussing them.
